hack/make-rules/test-cmd-util.sh

```diff
@@ -1062,26 +1062,16 @@ run_kubectl_apply_deployments_tests() {
   kube::test::get_object_assert replicasets "{{range.items}}{{$id_field}}:{{end}}" ''
   kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" ''
   # apply base deployment
-  kubectl apply -f hack/testdata/null-propagation/deployment-l1.yaml "${kube_flags[@]}"
+  kubectl apply -f hack/testdata/null-propagation/deployment.yaml "${kube_flags[@]}"
   # check right deployment exists
   kube::test::get_object_assert 'deployments my-depl' "{{${id_field}}}" 'my-depl'
   # check right labels exists
   kube::test::get_object_assert 'deployments my-depl' "{{.spec.template.metadata.labels.l1}}" 'l1'
   kube::test::get_object_assert 'deployments my-depl' "{{.spec.selector.matchLabels.l1}}" 'l1'
   kube::test::get_object_assert 'deployments my-depl' "{{.metadata.labels.l1}}" 'l1'
 
-  # apply new deployment with new template labels
```

Comment:
@kow3ns  i simplified the test to only use one YAML file (with null attributes). The motivation behind removing the second file is because two YAML files will break the new validation check introduced in this PR: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/50348/pull-kubernetes-unit/45541/

However, according to @liggitt, this breaks compatibility for old group's behavior. I think i am on the wrong track when adding the new validation check...